server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java

```diff
@@ -666,16 +677,28 @@ Releasable tryAcquireSnapshotDownloadPermits() {
         final int maxConcurrentSnapshotFileDownloads = getMaxConcurrentSnapshotFileDownloads();
         final boolean permitAcquired = maxSnapshotFileDownloadsPerNodeSemaphore.tryAcquire(maxConcurrentSnapshotFileDownloads);
         if (permitAcquired == false) {
-            logger.warn(
-                String.format(
-                    Locale.ROOT,
-                    "Unable to acquire permit to use snapshot files during recovery, "
-                        + "this recovery will recover index files from the source node. "
-                        + "Ensure snapshot files can be used during recovery by setting [%s] to be no greater than [%d]",
-                    INDICES_RECOVERY_MAX_CONCURRENT_SNAPSHOT_FILE_DOWNLOADS.getKey(),
-                    this.maxConcurrentSnapshotFileDownloadsPerNode
-                )
-            );
+            if (this.maxConcurrentIncomingRecoveries <= this.maxConcurrentSnapshotFileDownloadsPerNode) {
+                logger.warn(
+                    String.format(
+                        Locale.ROOT,
+                        "Unable to acquire permit to use snapshot files during recovery, "
+                            + "this recovery will recover index files from the source node. "
+                            + "Ensure snapshot files can be used during recovery by setting [%s] to be no greater than [%d]",
```

Comment:
Also here could we include the current values of the relevant settings in the message?

(and also perhaps convert the format string into a multi-line thing)

Reply:
Done